rust/vectordb/src/io/object_store.rs

```diff
@@ -142,6 +144,9 @@ impl ObjectStore for MirroringObjectStore {
     }
 
     async fn copy_if_not_exists(&self, from: &Path, to: &Path) -> Result<()> {
+        if !to.primary_only() {
+            self.secondary.copy(from, to).await?;
```

Comment:
same question.

Reply:
This is intentional, if we fail to write to secondary, we just fail the whole operation, because we can't guarantee the replication if secondary fails.

we write secondary first for consistency, as we list primary for discovering versions.